gravitee-node-kubernetes/src/main/java/io/gravitee/node/kubernetes/keystoreloader/KubernetesPemRegistryKeyStoreLoader.java

```diff
@@ -0,0 +1,197 @@
+
+    public static final String GRAVITEEIO_PEM_REGISTRY_LABEL = "gravitee.io/component";
+
+    public KubernetesPemRegistryKeyStoreLoader(KeyStoreLoaderOptions options, KubernetesClient kubernetesClient) {
+        super(options, kubernetesClient);
+        prepareLocations();
+    }
+
+    private void prepareLocations() {
+        if (options.getKeyStoreType().equalsIgnoreCase(CERTIFICATE_FORMAT_PEM_REGISTRY)) {
+            this.resources.put(CERTIFICATE_FORMAT_PEM_REGISTRY, null);
+        }
+    }
+
+    public static boolean canHandle(KeyStoreLoaderOptions options) {
+        return options.getKeyStoreType().equalsIgnoreCase(CERTIFICATE_FORMAT_PEM_REGISTRY);
+    }
+
+    @Override
+    protected Completable init() {
+        final List<Completable> locationObs = resources
+            .keySet()
+            .stream()
+            .map(location -> {
+                if (CERTIFICATE_FORMAT_PEM_REGISTRY.equals(location)) {
+                    String currentNamespace = KubernetesConfig.getInstance().getCurrentNamespace();
+                    return this.kubernetesClient.get(
+                            ResourceQuery
+                                .configMaps(currentNamespace)
+                                .labelSelector(
+                                    LabelSelector.equals(GRAVITEEIO_PEM_REGISTRY_LABEL, CERTIFICATE_FORMAT_PEM_REGISTRY.toLowerCase())
+                                )
+                                .build()
+                        )
+                        .flatMapCompletable(configMapList -> {
+                            List<ConfigMap> items = configMapList.getItems();
+                            if (items.isEmpty()) {
+                                return Completable.error(new RuntimeException("No pem registry found in the current namespace"));
+                            } else if (items.size() > 1) {
+                                return Completable.error(new RuntimeException("multiple pem registry is not supported"));
+                            } else {
+                                String newLocation = String.format(
+                                    "/%s/configmaps/%s",
+                                    currentNamespace,
+                                    items.get(0).getMetadata().getName()
+                                );
+                                this.resources.put(newLocation, ResourceQuery.<ConfigMap>from(newLocation).build());
+
+                                return this.loadKeyStore(items.get(0));
+                            }
+                        })
+                        .doOnComplete(() -> resources.remove(CERTIFICATE_FORMAT_PEM_REGISTRY));
+                }
+
+                return Completable.error(new RuntimeException(String.format("unsupported keystore locations %s", location)));
+            })
+            .toList();
+
+        return Completable.merge(locationObs).andThen(Completable.fromRunnable(this::refreshKeyStoreBundle));
```

Comment:
Could you please review this rx chain based on my previous comment and avoid this merge?